website/files/models/osfstorage.py

```diff
@@ -102,6 +103,35 @@ def move_under(self, destination_parent, name=None):
             raise exceptions.FileNodeCheckedOutError()
         return super(OsfStorageFileNode, self).move_under(destination_parent, name)
 
+    def check_in_or_out(self, user, checkout, save=False):
+        """
+        Updates self.checkout with the requesting user or None if user has permission
```

Comment:
Except for one thing: update this docstring with the parameter usage, since it's a little odd. https://cosdev.readthedocs.org/en/latest/osf/guidelines.html?highlight=docstring#docstrings